lib/webmock/http_lib_adapters/http_gem_adapter.rb

```diff
@@ -1,11 +1,10 @@
 begin
   require "http"
-  __http_gem_found__ = true
 rescue LoadError
-  __http_gem_found__ = false
+  # HTTP gem not found
 end
 
-if __http_gem_found__
+if defined?(HTTP) && defined?(HTTP::VERSION)
```

Comment:
You're right actually, I assumed it only checked the last thing in the chain (HTTP isn't defined in my project) but it seems that it's fine with any of the things being undefined.